### PR TITLE
Added phrase validation during edit

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#5559

*Description of changes:*
- phrase validation is now performed during edit (same as New page)
- the previous text message is replaced with error messages (same as New page)
- the error messages now include the required phrases
- increased error message line height to fix truncation

(See ticket for updated screenshots.)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: client/src/components/common/names-capture.vue [hunk 1]
```diff
@@ -85,12 +85,6 @@
           {{ mainMessage}}
         </v-col>
       </v-row>
-      <v-row v-if="entityPhraseRequired" class="my-1 py-0 colour-text">
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:88]: I replaced this with the pre-existing (and more general) validation mechanism.

File: client/src/components/common/names-capture.vue [hunk 2]
```diff
@@ -376,26 +373,7 @@ export default class NamesCapture extends Vue {
       }
       return outcome
     }
-    if (this.editMode) {
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:379]: I moved this below the block functions.

File: client/src/components/common/names-capture.vue [hunk 3]
```diff
@@ -461,6 +439,31 @@ export default class NamesCapture extends Vue {
       return true
     }
 
+    if (this.editMode) {
+      let outcome = true
+      if (designationAtEnd) {
+        if (!name1() || !name2() || !name3()) {
+          outcome = false
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:446]: This is new -- validate the names during edit.

File: client/src/components/common/names-capture.vue [hunk 4]
```diff
@@ -487,8 +490,8 @@ export default class NamesCapture extends Vue {
   }
   get mainMessage () {
     if (this.isAssumedName) {
-      return `${this.name}  is  too similar to a name already in use.  Please enter a new name such as your corporation
-  number`
+      return `${this.name} is too similar to a name already in use. Please enter a new name such as your
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:493]: I fixed this long line.

File: client/src/components/common/names-capture.vue [hunk 5]
```diff
@@ -652,19 +655,18 @@ export default class NamesCapture extends Vue {
     this.validate(true)
   }
   validatePhrases (choice: string) {
-    if (!this.editMode) {
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:655]: This now applies to both edit and new modes.

File: client/src/components/common/names-capture.vue [hunk 6]
```diff
@@ -652,19 +655,18 @@ export default class NamesCapture extends Vue {
...
-        if (this.entityPhraseChoices.some(phrase => name.startsWith(phrase))) {
-          this.messages[choice] = 'Your name must not begin with one of the required phrases'
-        }
+    if (this.entityPhraseRequired) {
+      const name = this.nameChoices[choice].toUpperCase()
+      if (this.entityPhraseChoices.every(phrase => {
+        phrase = phrase.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&')
+        return (name.search(new RegExp('(\\s)' + phrase + '(\\s|$)')) === -1)
+      })) {
+        this.messages[choice] = `A ${this.entityTypeText} name must include (but not start with)
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:664]: I updated the messages to include the entity type and entity phrases. This now applies to both edit and new modes.

**Review by scottrumsby — APPROVED** [review]
Seems to work properly now.  Looks good.

**Comment by severinbeauvais** [comment]
Tracey and Janis both OKed this change... Last reviewer, please merge. Thanks!

File: client/src/components/common/names-capture.vue [hunk 7]
```diff
@@ -652,25 +655,25 @@ export default class NamesCapture extends Vue {
...
     }
   }
 }
 </script>
 
-<style lang="sass">
-
+<style scoped lang="sass">
+::v-deep .v-messages__message
+  line-height: 14px !important
```
Comment (severinbeauvais) [inline client/src/components/common/names-capture.vue:678]: Line height _was_ 12px, same as the default font size here, but the bottom was cut off (ie, chars that dangled, like "g" and "y").

**Review by cameron-freshworks — APPROVED** [review]
Sorry, i reviewed but forgot to approve 😬

**Pull request merged**